### PR TITLE
Create DAO Marketplaces & Switch To Solana Wallet Adapter Everywhere

### PR DESCRIPTION
**Pull request by kespinola**
### Goal
Users are able to create a marketplace.

### Changes
- Refactor of connection to decouple storefronts from the wallet connector.
- Add marketplace provider.
- New page for creating a marketplace `/marketplace/new`
- Creating a marketplace loads config to ipfs, create auction house, and storefront v2.
- Move nft model to /nfts/new. Use router.back to simulate modal experience.
- Auto lookup storefront and upsert holaplex wallet record as soon as pubkey detected.
- Deprecate `window.solana` in favor of wallet adapter.
- Activate dao marketpace with `NEXT_PUBLIC_MARKETPLACE_ENABLED=true`

Screen Shot 2022-02-22 at 11 21 40 AM
Screen Shot 2022-02-22 at 11 22 05 AM
Screen Shot 2022-02-22 at 11 22 36 AM
Screen Shot 2022-02-22 at 11 37 14 AM

### CODE REVIEW

**Comment by sjc5** [comment]
@kespinola I addressed the outstanding comments from your [other branch](https://github.com/holaplex/holaplex/pull/224), but something in this refactor broke all wallet actions other than "connect".

Connecting / disconnecting is working as expected, but clicking "Update" at `/storefront/edit` or "Next" at `/marketplace/new`, for example, do not initiate any action.

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+nodejs 16.14.0
```
Comment (kristianeboe) [inline .tool-versions:1]: @kespinola what is this tool-versions? Maybe it's easier to keep track of node version in package.json?

File: pages/api/storefronts.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ const postArweaveStorefront = async (storefront: Storefront) => {
   const tx = await arweave.createTransaction({ data: stylesheet(storefront.theme) });
 
   tx.addTag('Content-Type', 'text/css');
+  // TODO: set for marketplace
```
Comment (kristianeboe) [inline pages/api/storefronts.ts:44]: Should this todo be fixed?

File: yarn.lock [hunk 3]
```diff
@@ -1787,6 +1787,48 @@
...
+
+"@metaplex-foundation/mpl-auction-house@file:../metaplex-program-library/auction-house/js":
+  version "0.0.2"
+  dependencies:
+    "@metaplex-foundation/beet" "0.0.7"
+    "@metaplex-foundation/beet-solana" "0.0.6"
+    "@metaplex-foundation/mpl-core" "0.0.3"
+    "@solana/web3.js" "^1.31.0"
+
+"@metaplex-foundation/mpl-auction-house@file:/Users/kyleespinola/Code/holaplex/metaplex-program-library/auction-house/js":
```
Comment (sjc5) [inline yarn.lock:1824]: Local file path

File: package.json [hunk 4]
```diff
@@ -29,14 +29,15 @@
     "@google/model-viewer": "^1.9.2",
     "@holaplex/js": "^4.3.2",
     "@holaplex/ui": "^0.0.26",
+    "@metaplex-foundation/mpl-auction-house": "file:/Users/kyleespinola/Code/holaplex/metaplex-program-library/auction-house/js",
```
Comment (sjc5) [inline package.json:32]: local file path

File: pages/storefront/edit.tsx [hunk 5]
```diff
@@ -296,7 +296,7 @@ export default function Edit() {
               </TabPane>
             </Tabs>
             <Row justify="end">
-              <Button disabled={submitting} loading={submitting}>
+              <Button disabled={submitting} loading={submitting} onClick={onSubmit}>
```
Comment (sjc5) [inline pages/storefront/edit.tsx:299]: I believe `onClick={onSubmit}` should be `htmlType="submit" to avoid bypassing the form validation

**Review by sjc5 — APPROVED** [review]
Create mrktplace flow isn't 100% like @damianr's mockups due to some complications with the ant upload components.  But it's looking good, and Damian said he is happy with this.  Fully tested, everything works great.  Ready to merge.

**Comment by kespinola** [comment]
These are the items I tested:

- [ ] Mint NFT
- [ ] Edit storefront
- [ ] View profile
- [ ] Create marketplace
- [ ] Disconnect
- [ ] Deep link connect and visit storefront new, storefront edit, marketplace new, mint nft

**Comment by kespinola** [comment]
Screen Shot 2022-03-01 at 7 15 57 AM

The plus button on mint nft is off to the left.

Screen Shot 2022-03-01 at 7 19 09 AM

No X on clear attribute field

**Comment by sjc5** [comment]
> Screen Shot 2022-03-01 at 7 15 57 AM
> 
> The plus button on mint nft is off to the left.
> 
> Screen Shot 2022-03-01 at 7 19 09 AM
> 
> No X on clear attribute field

These style bugs aren't specific to this PR and exist in Holaplex.com currently

**Pull request merged**